lib/poller.js

```diff
@@ -86,22 +163,36 @@ function start (__settings) {
 
-  setInterval(() => pi().executeTrades(), TRADE_INTERVAL)
-  setInterval(() => cashOutTx.monitorLiveIncoming(settings(), false, coinFilter), LIVE_INCOMING_TX_INTERVAL)
-  setInterval(() => cashOutTx.monitorStaleIncoming(settings(), false, coinFilter), INCOMING_TX_INTERVAL)
+  addToQueue(pi().executeTrades, TRADE_INTERVAL, schema, QUEUE.FAST)
+  addToQueue(cashOutTx.monitorLiveIncoming, LIVE_INCOMING_TX_INTERVAL, schema, QUEUE.FAST, settings, false, coinFilter)
+  addToQueue(cashOutTx.monitorStaleIncoming, INCOMING_TX_INTERVAL, schema, QUEUE.FAST, settings, false, coinFilter)
   if (!_.isEmpty(coinFilter)) {
-    setInterval(() => cashOutTx.monitorLiveIncoming(settings(), true, coinFilter), LIVE_INCOMING_TX_INTERVAL_FILTER)
-    setInterval(() => cashOutTx.monitorStaleIncoming(settings(), true, coinFilter), INCOMING_TX_INTERVAL_FILTER)
+    addToQueue(cashOutTx.monitorLiveIncoming, LIVE_INCOMING_TX_INTERVAL_FILTER, schema, QUEUE.FAST, settings, true, coinFilter)
+    addToQueue(cashOutTx.monitorStaleIncoming, INCOMING_TX_INTERVAL_FILTER, schema, QUEUE.FAST, settings, true, coinFilter)
   }
-  setInterval(() => cashOutTx.monitorUnnotified(settings()), UNNOTIFIED_INTERVAL)
-  setInterval(() => cashInTx.monitorPending(settings()), PENDING_INTERVAL)
-  setInterval(() => pi().sweepHd(), SWEEP_HD_INTERVAL)
-  setInterval(() => pi().pong(), PONG_INTERVAL)
-  setInterval(() => pi().clearOldLogs(), LOGS_CLEAR_INTERVAL)
-  setInterval(() => notifier.checkNotification(pi()), CHECK_NOTIFICATION_INTERVAL)
-  setInterval(initialSanctionsDownload, SANCTIONS_INITIAL_DOWNLOAD_INTERVAL)
-  setInterval(updateAndLoadSanctions, SANCTIONS_UPDATE_INTERVAL)
-  setInterval(updateCoinAtmRadar, RADAR_UPDATE_INTERVAL)
+  addToQueue(cashOutTx.monitorUnnotified, UNNOTIFIED_INTERVAL, schema, QUEUE.FAST, settings)
+  addToQueue(cashInTx.monitorPending, PENDING_INTERVAL, schema, QUEUE.FAST, settings)
+  addToQueue(pi().sweepHd, SWEEP_HD_INTERVAL, schema, QUEUE.FAST, settings)
+  addToQueue(pi().pong, PONG_INTERVAL, schema, QUEUE.FAST)
+  addToQueue(pi().clearOldLogs, LOGS_CLEAR_INTERVAL, schema, QUEUE.SLOW)
+  addToQueue(notifier.checkNotification, CHECK_NOTIFICATION_INTERVAL, schema, QUEUE.FAST, pi)
+  addToQueue(initialSanctionsDownload, SANCTIONS_INITIAL_DOWNLOAD_INTERVAL, schema, QUEUE.SLOW)
+  addToQueue(updateAndLoadSanctions, SANCTIONS_UPDATE_INTERVAL, schema, QUEUE.SLOW)
+  addToQueue(updateCoinAtmRadar, RADAR_UPDATE_INTERVAL, schema, QUEUE.SLOW)
 }
 
-module.exports = { start, reload }
+function setup (schemasToAdd = [], schemasToRemove = []) {
+  // clear callback array for each schema in schemasToRemove and clear cached variables
+  _.forEach(schema => {
+    const callbacks = schemaCallbacks.get(schema)
+    _.forEach(clearInterval, callbacks)
+    schemaCallbacks.delete(schema)
+    cachedVariables.del(schema)
+  }, schemasToRemove)
+
+  return initializeEachSchema(schemasToAdd)
+}
+
+const getActiveSchemas = () => Array.from(schemaCallbacks.keys())
```

Comment:
Is there any nuance which requires the array creation to be like this, instead of something like `_.keys(schemaCallbacks)`?

Reply:
Yes, the fact I'm using a Map instead of an object. I think we can change it to an object since the functionality I wanted with Map isn't necessary anymore